samples/bluetooth/nrf_dm/src/main.c

```diff
@@ -0,0 +1,319 @@
+{
+	struct dm_request req;
+
+	if (peer_supported_test(info->addr)) {
+		bt_addr_le_copy(&req.bt_addr, info->addr);
+		req.role = DM_ROLE_REFLECTOR;
+		req.ranging_mode = ranging_mode;
+		req.access_address = peer_access_address_get();
+		req.start_delay_us = 0;
+
+		dm_request_add(&req);
+	}
+
+	mfg_data.access_address = sys_cpu_to_le32(peer_access_address_new());
+	k_work_submit(&adv_work);
+}
+
+const static struct bt_le_ext_adv_cb adv_cb = {
+	.scanned = adv_scanned_cb,
+};
+
+static int adv_start(void)
+{
+	int err;
+	struct bt_le_ext_adv_start_param ext_adv_start_param = {0};
+
+	if (adv) {
+		err = bt_le_ext_adv_stop(adv);
+		if (err) {
+			printk("Failed to stop extended advertising  (err %d)\n", err);
+			return err;
+		}
+			err = bt_le_ext_adv_delete(adv);
+		if (err) {
+			printk("Failed to delete advertising set  (err %d)\n", err);
+			return err;
+		}
+	}
+
+	err = bt_le_ext_adv_create(adv_param, &adv_cb, &adv);
+	if (err) {
+		printk("Failed to create advertising set (err %d)\n", err);
+		return err;
+	}
+
+	err = bt_le_ext_adv_set_data(adv, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (err) {
+		printk("Failed setting adv data (err %d)\n", err);
+		return err;
+	}
+
+	err = bt_le_ext_adv_start(adv, &ext_adv_start_param);
+	if (err) {
+		printk("Failed to start extended advertising  (err %d)\n", err);
+		return err;
+	}
+
+	return err;
+}
+
+static void adv_work_handle(struct k_work *item)
+{
+	adv_start();
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	adv_param = &adv_param_noconn;
+	k_work_submit(&adv_work);
+
+	dk_set_led_on(CON_STATUS_LED);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	adv_param = &adv_param_conn;
+	k_work_submit(&adv_work);
+
+	dk_set_led_off(CON_STATUS_LED);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int bt_sync_init(void)
+{
+	/* Synchronisation is based on advertising and scanning modes.
+	 * It occurs when SCAN_REQ and SCAN_RESP packets are exchanged.
+	 */
+
+	int err;
+
+	printk("DM Bluetooth LE Synchronization initialization\n");
+
+	mfg_data.company_code = sys_cpu_to_le16(COMPANY_CODE);
+	mfg_data.support_dm_code = sys_cpu_to_le32(SUPPORT_DM_CODE);
+	mfg_data.access_address = sys_cpu_to_le32(peer_access_address_new());
+
+	err = adv_start();
+	if (err) {
+		printk("Failed to start advertising  (err %d)\n", err);
+		return err;
+	}
+
+	bt_scan_init(&scan_init);
+	bt_scan_cb_register(&scan_cb);
+
+	err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_MANUFACTURER_DATA, &scan_mfg_data);
+	if (err) {
+		printk("Scanning filters cannot be set (err %d)\n", err);
+		return err;
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_MANUFACTURER_DATA_FILTER, false);
+	if (err) {
+		printk("Filters cannot be turned on (err %d)\n", err);
+		return err;
+	}
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return err;
+	}
```

Comment:
Consider defining a dedicated function for scanning so it's isolated like the advertising part

Reply:
Good point. Done.